app/models/user.rb

```diff
@@ -17,6 +17,10 @@ class User < ActiveRecord::Base
   validates :last_name, :length => {minimum: 2}
   validate :sensor_codes_string_contains_only_valid_sensors
   validates_presence_of :address, :email, :zip_code
+  validates_format_of :zip_code,
+                      with: /\A\d{5}-\d{4}|\A\d{5}\z/,
+                      message: "should be 12345 or 12345-1234",
+                      allow_blank: true
```

Comment:
Awesome! Looks great, I wouldn't have thought to validate for the optional extra digits.